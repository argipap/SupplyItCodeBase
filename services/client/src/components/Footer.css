.footer {
    width: 100%;
    height: 50px;
    line-height: 50px;
    margin-top: 50px;
    background-color: white;
}