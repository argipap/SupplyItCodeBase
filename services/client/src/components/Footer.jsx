import React from 'react';
import PropTypes from 'prop-types';
import './Footer.css';
import { Container, Col, Row } from 'react-bootstrap';
import { HashLink as Link } from 'react-router-hash-link';


const Footer = (props) => {

	let footer = (
		<ul className="footer-list">
			<li>
				<Link smooth to="/">Αρχική</Link>
			</li>
			<li>
				<Link smooth to="/#howitworks">Πως δουλεύει</Link>
			</li>
			<li>
				<Link smooth to="/#getstarted">Δοκιμάστε</Link>
			</li>
			<li>
				<Link to="/BecomeSupplier">Γίνετε προμηθευτής</Link>
			</li>
			<li>
				<Link to="/login">Σύνδεση</Link>
			</li>
		</ul>

	)

	if (props.isAuthenticated()) {
		footer = (
			<ul className="footer-list">
			<li>
				<Link smooth to="/">Αρχική</Link>
			</li>
			<li>
				<Link to="/status">Status</Link>
			</li>
			<li>
				<Link smooth to="/#howitworks">Πως δουλεύει</Link>
			</li>
			<li>
				<Link to="/logout">Αποσύνδεση</Link>
			</li>
		</ul>
		)
	}

	return (

		<Container className="">
			<div className="footer-trust_top">
				<p>Μας προτείνουν</p>
				<div className="trustlogos" />
			</div>
			<hr />
			<Row className="footer-main">
				<Col xs={12} md={3}>
					<h4>Περισσότερα</h4>
					{footer}
				</Col>
				<Col xs={12} md={6} className="footer-middle">
					<h4>Εγγραφείτε στο Newsletter μας</h4>
					<p>Εγγραφείτε για να λαμβάνετε τα νέα της πλατφόρμας και άλλα πολλά.</p>

					<form className="hero-cta__form">
						<div className="elcontainer">
							<div className="inner-wrap submit-container">
								<div className="hero-cta__input-wrap">
									<input
										type="email"
										name="data[email]"
										required=""
										id="hero-cta-email"
										placeholder="Το email σας"
									/>
								</div>
								<button type="submit" className="emailsignup">
									Εγγραφή
							</button>
							</div>
						</div>
					</form>
				</Col>

				<Col xs={12} md={3} className="footer-right">
					<div className="footerbox">
						<h4>Επικοινωνία</h4>
						<p>Είμαστε εδώ για εσάς</p>
						<ul className="footer-list">
							<li>Σταθερό</li>
							<li className="phone">+30 88888888</li>
							<li>Κινητό</li>
							<li className="phone">+30 23123123</li>
						</ul>
					</div>
				</Col>
			</Row>

			<Row className="footer-bottom">
				<Col lg={3} md={3} className="footer-copyright">
					<p>©2019 Supplyit.gr</p>
				</Col>

				<Col md={9} xs={12} className="footer-partners">
					<img src="https://www.authenticireland.com/wp-content/uploads/2018/04/Bottom_Trust2.png" alt="" />
				</Col>
			</Row>
		</Container>
	)
};

Footer.propTypes = {
    logoutUser: PropTypes.func.isRequired,
    isAuthenticated: PropTypes.func.isRequired
};

export default Footer;
