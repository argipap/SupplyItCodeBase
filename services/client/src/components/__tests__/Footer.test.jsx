import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import Footer from '../Footer';

test('Footer renders properly', () => {
    const wrapper = mount(<Footer/>);
    const element = wrapper.find('div.footer-copyright');
    expect(element.length).toBe(1);
    expect(element.text()).toBe('Â©2019 Supplyit.gr');
});

test('Footer renders a snapshot properly', () => {
    const tree = renderer.create(<Footer/>).toJSON();
    expect(tree).toMatchSnapshot();
});