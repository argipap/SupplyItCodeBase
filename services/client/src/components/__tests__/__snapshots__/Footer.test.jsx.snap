// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders a snapshot properly 1`] = `
<div
  className="container"
>
  <div
    className="footer-trust_top"
  >
    <p>
      Μας προτείνουν
    </p>
    <div
      className="trustlogos"
    />
  </div>
  <hr />
  <div
    className="footer-main row"
  >
    <div
      className="col-md-3 col-12"
    >
      <h4>
        Περισσότερα
      </h4>
      <ul
        className="footer-list"
      >
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            Αρχική
          </a>
        </li>
        <li>
          <a
            href="/#howitworks"
            onClick={[Function]}
          >
            Πως δουλεύει
          </a>
        </li>
        <li>
          <a
            href="/#getstarted"
            onClick={[Function]}
          >
            Δοκιμάστε
          </a>
        </li>
        <li>
          <a
            href="/BecomeSupplier"
            onClick={[Function]}
          >
            Γίνετε προμηθευτής
          </a>
        </li>
        <li>
          <a
            href="/login"
            onClick={[Function]}
          >
            Σύνδεση
          </a>
        </li>
      </ul>
    </div>
    <div
      className="footer-middle col-md-6 col-12"
    >
      <h4>
        Εγγραφείτε στο Newsletter μας
      </h4>
      <p>
        Εγγραφείτε για να λαμβάνετε τα νέα της πλατφόρμας και άλλα πολλά.
      </p>
      <form
        className="hero-cta__form"
      >
        <div
          className="elcontainer"
        >
          <div
            className="inner-wrap submit-container"
          >
            <div
              className="hero-cta__input-wrap"
            >
              <input
                id="hero-cta-email"
                name="data[email]"
                placeholder="Το email σας"
                required=""
                type="email"
              />
            </div>
            <button
              className="emailsignup"
              type="submit"
            >
              Εγγραφή
            </button>
          </div>
        </div>
      </form>
    </div>
    <div
      className="footer-right col-md-3 col-12"
    >
      <div
        className="footerbox"
      >
        <h4>
          Επικοινωνία
        </h4>
        <p>
          Είμαστε εδώ για εσάς
        </p>
        <ul
          className="footer-list"
        >
          <li>
            Σταθερό
          </li>
          <li
            className="phone"
          >
            +30 88888888
          </li>
          <li>
            Κινητό
          </li>
          <li
            className="phone"
          >
            +30 23123123
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="footer-bottom row"
  >
    <div
      className="footer-copyright col-lg-3 col-md-3"
    >
      <p>
        ©2019 Supplyit.gr
      </p>
    </div>
    <div
      className="footer-partners col-md-9 col-12"
    >
      <img
        alt=""
        src="https://www.authenticireland.com/wp-content/uploads/2018/04/Bottom_Trust2.png"
      />
    </div>
  </div>
</div>
`;
